ct/bookstack.sh

```diff
@@ -63,33 +63,40 @@ if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_v
   msg_info "Stopping Apache2"
   systemctl stop apache2
   msg_ok "Services Stopped"
-
   msg_info "Updating ${APP} to ${RELEASE}"
-  cp /opt/bookstack/.env /opt/.env
+  mv /opt/bookstack /opt/bookstack-backup 
   wget -q "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
   unzip -q v${RELEASE}.zip
-  mv BookStack-${RELEASE} /opt/bookstack
-  mv /opt/.env /opt/bookstack/.env
+  mv BookStack-${RELEASE}/ /opt/bookstack
+  cp /opt/bookstack-backup/.env /opt/bookstack/.env
+  cp -r /opt/bookstack-backup/public/uploads/ /opt/bookstack/public/uploads
+  cp -r /opt/bookstack-backup/storage/uploads/ /opt/bookstack/storage/uploads
+  cp -r /opt/bookstack-backup/themes/ /opt/bookstack/themes
   cd /opt/bookstack
   COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev  &>/dev/null
   php artisan key:generate --force &>/dev/null
   php artisan migrate --force &>/dev/null
+  chown www-data:www-data -R /opt/bookstack /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads /opt/bookstack/storage 
+  chmod -R 755 /opt/bookstack /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads /opt/bookstack/storage 
+  chmod -R 775 /opt/bookstack/storage /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads
+  chmod -R 640 /opt/bookstack/.env 
   echo "${RELEASE}" >/opt/${APP}_version.txt
   msg_ok "Updated ${APP}"
-
   msg_info "Starting Apache2"
   systemctl start apache2
   msg_ok "Started Apache2"
-
   msg_info "Cleaning Up"
-  rm -rf v${RELEASE}.zip
-  msg_ok "Cleaned"
+  cd /root/
```

Comment:
Do you know that the current working directory is `/root/` when starting the update script? What if the script is started in some other directory?

Reply:
I change it so it is ensured the files we clean are always at /root/. The need is, I switch to /opt/bookstack before the cleanup.